includes/admin/class-wc-rest-upe-flag-toggle-controller.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Class WC_REST_Upe_Flag_Toggle_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for UPE feature flag.
+ */
+class WC_REST_Upe_Flag_Toggle_Controller extends WP_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
```

Comment:
For consistency with the other controllers.

```suggestion
```

Reply:
The other controllers are extending from `WC_Payments_REST_Controller` (where the `namespace` is defined), this extends from `WP_REST_Controller`, since there's no need to depend from `WC_Payments_REST_Controller` and the `WC_Payments_API_Client` class in the constructor.

Reply:
Oh, nice! I didn't notice that. Kudos for keeping an eye on what's not needed!

I wonder if it'd be nicer to extend it anyway for consistency (using the same default namespace, permission check, maybe something more in the future).

The API client is instantiated only once and then just passed to each constructor so there shouldn't be a performance hit - it's probably just a matter of code cleanliness.

Not saying I'm right - just my $.02.

Reply:
I also considered it initially, but I thought that not having the extra dependency would make things easier also from a testing perspective, so that we don't need to mock stuff we don't need 🤷

Reply:
Yep, solid reasoning!